### PR TITLE
Updated python script to export BESTEST data

### LINKED ISSUE #3886: Update BESTEST python script
In order to add MBL results to ASHRAE 140-2023 results portal, we need to update the Python script, to export new results:
- [ ] export the results that are used for generating the plots, to the database
- [ ] add new variable results, 
     - solar radiation: annual incident (case 600), unshaded annual transmitted (case 600/South, case 660/South, case 670/South, case 620/South), shaded annual transmitted (case 610/South, case 630/West)
     - Sky temperature output: case 600, annual hourly average, minimum (degC, month-day: hour), maximum (degC, month-day: hour)
     - Monthly conditioned zone loads (case 600 and 900): total heating, total cooling, minimum (degC, month-day: hour), maximum (degC, month-day: hour)
     - Specific data hourly output

### PR DESCRIPTION
**Pull request by JayHuLBL**
This closes #3886.

### CODE REVIEW

File: Buildings/ThermalZones/Detailed/Validation/BESTEST/Cases6xx/Case600.mo [hunk 1]
```diff
@@ -73,6 +73,47 @@ model Case600 "Case 600FF, but with dual-setpoint for heating and cooling"
   Buildings.Controls.OBC.CDL.Reals.MovingAverage PCoo(delta=3600)
     "Hourly averaged cooling power"
     annotation (Placement(transformation(extent={{-20,-8},{-12,0}})));
+  Modelica.Blocks.Sources.RealExpression hGloHor(
```
Comment (mwetter) [inline Buildings/ThermalZones/Detailed/Validation/BESTEST/Cases6xx/Case600.mo:76]: Always add the unit to these expressions.

File: Buildings/ThermalZones/Detailed/Validation/BESTEST/Cases6xx/Case600.mo [hunk 2]
```diff
@@ -73,6 +73,47 @@ model Case600 "Case 600FF, but with dual-setpoint for heating and cooling"
...
+    y_start=0) "Annual global horizontal solar irradiance"
+    annotation (Placement(transformation(extent={{74,26},{82,34}})));
+  replaceable Modelica.Blocks.Sources.RealExpression hGloSou(
+    y=roo.conExtWinRad[1].HDir + roo.conExtWinRad[1].HDif)
+    "South total solar irradiance"
+    annotation (Placement(transformation(extent={{40,0},{60,20}})));
+  Modelica.Blocks.Continuous.Integrator gloSou(
+    k=1,
+    initType=Modelica.Blocks.Types.Init.InitialState,
+    y_start=0) "Annual south total solar irradiance"
```
Comment (mwetter) [inline Buildings/ThermalZones/Detailed/Validation/BESTEST/Cases6xx/Case600.mo:92]: This should be "global" not "total" in the comment. Also above for `hGloSou`

File: Buildings/ThermalZones/Detailed/Validation/BESTEST/Cases6xx/Case600.mo [hunk 3]
```diff
@@ -73,6 +73,47 @@ model Case600 "Case 600FF, but with dual-setpoint for heating and cooling"
...
+    annotation (Placement(transformation(extent={{74,6},{82,14}})));
+  Modelica.Blocks.Continuous.Integrator traSol(
+    k=1,
+    initType=Modelica.Blocks.Types.Init.InitialState,
+    y_start=0) "Annual transmitted solar irradiance"
+    annotation (Placement(transformation(extent={{-40,86},{-32,94}})));
+  replaceable Modelica.Blocks.Sources.RealExpression hTra(
+    y=(roo.conExtWinRad[1].QTraDif_flow + roo.conExtWinRad[1].QTraDir_flow)/(roo.conExtWinRad[1].AWin))
+    "Transmistted solar radiation"
```
Comment (mwetter) [inline Buildings/ThermalZones/Detailed/Validation/BESTEST/Cases6xx/Case600.mo:101]: typo

File: Buildings/ThermalZones/Detailed/Validation/BESTEST/Cases6xx/Case620.mo [hunk 4]
```diff
@@ -63,5 +75,13 @@ October 6, 2011, by Michael Wetter:<br/>
 First implementation.
 </li>
 </ul>
-</html>"));
+</html>"),
+    Diagram(graphics={Text(
+          extent={{-74,104},{8,96}},
+          textColor={28,108,200},
+          textString="For the case, it is the transmitted through west window."),
```
Comment (mwetter) [inline Buildings/ThermalZones/Detailed/Validation/BESTEST/Cases6xx/Case620.mo:82]: You mean "transmitted solar radiation"?

File: Buildings/Resources/src/ThermalZones/Detailed/Validation/BESTEST/simulateAndPlot.py [hunk 5]
```diff
@@ -412,6 +454,219 @@ def _find_data_bin(dataSet):
         results[i] = '{}'.format(int(counter))
     return results
 
+def export_data(data_set):
+    KWh = 3600*1e3
```
Comment (mwetter) [inline Buildings/Resources/src/ThermalZones/Detailed/Validation/BESTEST/simulateAndPlot.py:458]: For consistency I would use lower case `kWh` as kilo is abbreviated lower case.

File: Buildings/Resources/src/ThermalZones/Detailed/Validation/BESTEST/simulateAndPlot.py [hunk 6]
```diff
@@ -412,6 +454,219 @@ def _find_data_bin(dataSet):
...
+    results = list()
+    for k in range(len(data_set)):
+        data = data_set[k]
+        if 'FF' in data['case']:
+            temp = {'case': data['case']}
+            for varVal in data['result']:
+                if varVal['variable'] == 'TRooHou.y':
+                    annual = varVal['value']
+                    Feb1 = annual[745:769]
+                    Jul14 = annual[4657:4681]
```
Comment (mwetter) [inline Buildings/Resources/src/ThermalZones/Detailed/Validation/BESTEST/simulateAndPlot.py:468]: Is this typo of code robust, e.g., does the `745:769` change if Dymola (or Optimica?) writes more or fewer output lines, perhaps when a future version changed the solver or has more events?
Perhaps you should first search at what line Feb 1 starts and ends, and then use this index.

**Review by mwetter — CHANGES_REQUESTED** [review]
Please see inline comments.

File: Buildings/ThermalZones/Detailed/Validation/BESTEST/Cases6xx/Case600.mo (hunk 1 above)
Comment (JayHuLBL) [reply Buildings/ThermalZones/Detailed/Validation/BESTEST/Cases6xx/Case600.mo:76]: added.

File: Buildings/ThermalZones/Detailed/Validation/BESTEST/Cases6xx/Case600.mo (hunk 2 above)
Comment (JayHuLBL) [reply Buildings/ThermalZones/Detailed/Validation/BESTEST/Cases6xx/Case600.mo]: changed.

File: Buildings/Resources/src/ThermalZones/Detailed/Validation/BESTEST/simulateAndPlot.py (hunk 5 above)
Comment (JayHuLBL) [reply Buildings/Resources/src/ThermalZones/Detailed/Validation/BESTEST/simulateAndPlot.py]: changed.

File: Buildings/Resources/src/ThermalZones/Detailed/Validation/BESTEST/simulateAndPlot.py (hunk 6 above)
Comment (JayHuLBL) [reply Buildings/Resources/src/ThermalZones/Detailed/Validation/BESTEST/simulateAndPlot.py:468]: The function `def _extract_data(matFile, relVal):` from line 205 to line 243 converted the data point to (0~8760), so the start and end indices can be fixed here and it is flexible to handle more or less output lines.

```
# --------------------------------------------------------------------------------------------------
# Extract needed time series data from modelica simulation results
# --------------------------------------------------------------------------------------------------
def _extract_data(matFile, relVal):
    ...
```

**Review by mwetter — APPROVED** [review]

**Comment by mwetter** [comment]
@JayHuLBL : can you please check why the reference results don't match and update if needed. Note that the CI tests are using Dymola 2024x Refresh 1.

**Pull request merged**